bionic/protocols.py

```diff
@@ -197,6 +198,39 @@ def __repr__(self):
         return f"{self.__class__.__name__}(...)"
 
 
+class JsonProtocol(BaseProtocol):
+    """
+    Decorator indicating that an entity's values can be serialized to the
+    ``JSON`` format.
+    """
+
+    def get_fixed_file_extension(self):
+        return "json"
+
+    def validate(self, value):
+        if value is None:
+            return
+        if isinstance(value, list):
+            for elem in value:
+                self.validate(elem)
+            return
+        if isinstance(value, dict):
+            for key, elem in value.items():
+                assert isinstance(key, str)
+                self.validate(elem)
+            return
+
+        assert isinstance(value, (int, float, str, bool))
+
+    def write(self, value, path):
+        with path.open("w", encoding="utf-8") as file_:
+            json.dump(value, file_, ensure_ascii=False)
```

Comment:
It doesn't escape non-ascii characters in the output. Now that users can read cached artifacts directly, it would be nice if they can see the characters as they are instead of escaped values.

Reply:
Make sense to me!